Assets/Scripts/ScreenTapping.cs

```diff
@@ -9,14 +9,35 @@
 using UnityEngine.EventSystems;
 using System;
 
-public class ScreenTapping : MonoBehaviour, IPointerDownHandler
+public class ScreenTapping : MonoBehaviour, IPointerDownHandler, IBeginDragHandler, IDragHandler, IEndDragHandler
 {
     [SerializeField]
     [Tooltip("The prefab to instantiate for ScreenTapping.")]
     GameObject prefabScreenTapping;
 
     Vector2 localPoint;
 
+    public void OnBeginDrag(PointerEventData eventData)
+    {
+        RectTransformUtility.ScreenPointToLocalPointInRectangle(GetComponent<RectTransform>(), eventData.position, eventData.pressEventCamera, out localPoint);
+        GameObject tappingAnim = Instantiate(prefabScreenTapping, transform, false);
+        tappingAnim.transform.localPosition = localPoint;
+    }
+
+    public void OnDrag(PointerEventData eventData)
```

Comment:
There is far too much repeated code. Try to avoid repeating yourself when writing code. You can resolve this by putting the repeated code into its own method, then calling that method whenever you need it.